### PR TITLE
Add float16 comparison test

### PR DESCRIPTION
**Pull request by take-cheeze**
Resolves: https://github.com/chainer/chainer/projects/2#card-18795691

### CODE REVIEW

File: chainerx_cc/chainerx/float16_test.cc [hunk 1]
```diff
@@ -235,5 +235,26 @@ TEST(NativeFloat16Test, Float16DivideI) {
     }
 }
 
+TEST(NativeFloat16Test, FloatComparison) {
+    for (Float16 x : GetFloat16Values()) {
+        for (Float16 y : GetFloat16Values()) {
+#define run_compare_check(op)                                                \
```
Comment (hvy) [inline chainerx_cc/chainerx/float16_test.cc]: Could you follow the style in this test (also regarding the naming conventions)? https://github.com/chainer/chainer/blob/master/chainerx_cc/chainerx/routines/type_util_test.cc

File: chainerx_cc/chainerx/float16_test.cc [hunk 2]
```diff
@@ -235,5 +235,26 @@ TEST(NativeFloat16Test, Float16DivideI) {
...
+    do {                                                                     \
+        EXPECT_EQ(static_cast<double>(x) op static_cast<double>(y), x op y); \
+        EXPECT_EQ(static_cast<double>(y) op static_cast<double>(x), y op x); \
+    } while (false)
+
+            run_compare_check(==);
+            run_compare_check(!=);
+            run_compare_check(<);
+            run_compare_check(>);
```
Comment (hvy) [inline chainerx_cc/chainerx/float16_test.cc]: This and the line above are invoking cpplint errors. Maybe `NOLINT` is fine. You can by the way check the CI results, in this case Travis that reported it.

**Comment by hvy** [comment]
It basically LGTM but I wrote some comments if you could take a look. And thanks for the PR!

**Comment by take-cheeze** [comment]
@hvy Thanks for the review!
I've fixed it and travis seems to be passed!

File: chainerx_cc/chainerx/float16_test.cc [hunk 3]
```diff
@@ -235,5 +235,26 @@ TEST(NativeFloat16Test, Float16DivideI) {
...
 }
 
+TEST(NativeFloat16Test, FloatComparison) {
+    for (Float16 x : GetFloat16Values()) {
+        for (Float16 y : GetFloat16Values()) {
+#define CHECK_COMPARISION_OPERATOR(op)                                       \
+    {                                                                        \
+        EXPECT_EQ(static_cast<double>(x) op static_cast<double>(y), x op y); \
+        EXPECT_EQ(static_cast<double>(y) op static_cast<double>(x), y op x); \
+    };
```
Comment (hvy) [inline chainerx_cc/chainerx/float16_test.cc]: A nitpick.
```suggestion
    }
```

**Comment by hvy** [comment]
Thanks for the quick fix. I left a really minor comment.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
LGTM!

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Oh, could you take a look at the Jenkins failures? There's one in particular `TEST=CHAINERX_misc` which is a style error.

**Comment by hvy** [comment]
Jenkins, test this please.

**Pull request merged**